src/core/server/plugins/plugins_system.test.ts

```diff
@@ -126,7 +126,7 @@ test('`setupPlugins` throws if plugins have circular required dependency', async
   pluginsSystem.addPlugin(createPlugin('depends-on-2', { required: ['depends-on-1'] }));
 
   await expect(pluginsSystem.setupPlugins(setupDeps)).rejects.toMatchInlineSnapshot(
-    `[Error: Topological ordering of plugins did not complete, these edges could not be ordered: [["depends-on-1",{}],["depends-on-2",{}]]]`
+    `[Error: Topological ordering of plugins did not complete, cyclic dependency detected between: ["depends-on-1","depends-on-2"]]`
```

Comment:
Can we guarantee that they have circular dependencies? Probably both have missing deps.

Reply:
This is the pluginSystem sorting algorithm. Unsorted Kahn Graph is pretty much a cyclic dependency yes. ( I only changed the error message here)